sql/select.go

```diff
@@ -121,19 +120,14 @@ func (p *planner) expandSubqueries(stmt parser.Statement) error {
 // the query (we currently only support covering indexes) and selects the
 // "best" index from that set. The cost model based on keys per row, key size
 // and number of index elements used.
-func (p *planner) selectIndex(s *scanNode) (planNode, error) {
-	if s.desc == nil || s.filter == nil {
-		// No table or where-clause.
+func (p *planner) selectIndex(s *scanNode, ordering []int) (planNode, error) {
+	if s.desc == nil || (s.filter == nil && ordering == nil) {
```

Comment:
i might not be seeing it, but it seems that `ordering` can never be nil since it's only ever passed in from `sort.Ordering()` above `sortNode`s are never constructed with a nil slice.